design/object-store-realm.md

```diff
@@ -0,0 +1,214 @@
+
+### Prerequisites
+
+When a zone is being created on another Ceph cluster, an RGW in the master zone must have an external endpoint exposed that can be reached by the other cluster.
+
+### Ceph multi-site object store data model
+
+For reference, here is a description of the underlying Ceph Multisite data model.
+
+```
+A cluster has one or more realms
+
+A realm spans one or more clusters
+A realm has one or more zonegroups
+A realm has one master zonegroup
+A realm defined in another cluster is replicated with the pull command
+
+A zonegroup has one or more zones
+A zonegroup has one master zone
+A zonegroup spans one or more clusters
+A zonegroup defined in another cluster is replicated with the pull command
+A zonegroup defines a namespace for object IDs unique across its zones
+Zonegroup metadata is replicated to other zonegroups in the realm
+
+A zone belongs to one cluster
+A zone has a set of pools that store the user and object metadata and object data
+Zone data and metadata is replicated to other zones in the zonegroup
+A master zone needs to be created for secondary zones to pull from to replicate across zones
+```
+
+## Object Store Multisite Walkthrough
+
+### Creating a Realm & Zonegroups
+
+#### Config
+
+When the storage admin is ready to create a realm and zonegroups for object storage, the admin will specify his/her desired settings in the Realm section on the configuration file.
+
+This example `object-store-realm.yaml`, names a realm with 2 zonegroups.
+```yaml
+apiVersion: ceph.rook.io/v1alpha1
+kind: CephObjectStoreRealm
+metadata:
+  name: my-realm
+  namespace: rook-ceph
+spec:
+  bootstrap-realm:
+    name: my-realm
+    zonegroups:
+      my-zonegroup-a:
+        is-master: true
+        master-zone: zone-z
+      my-zonegroup-b:
+        is-master: false
+        master-zone: zone-y
+```
+
+Now create the object store realm.
+```bash
+kubectl create -f object-store-realm.yaml
+```
+
+#### Steps
+
+At this point the Rook operator recognizes that a new object store realm resource needs to be configured. The operator will create all of the resources to start the object store realm.
+
+1. A Ceph realm is created if it does not exist already
+2. A Ceph master zonegroup is created in the realm if it does not exist already
+3. Any non-master Ceph zonegroup is created in the realm if it does not exist already
+4. A master zone is created for the zonegroup.
+
+After these steps the admin should start up object stores configured with the same zone, zonegroup, and realm as configured (see next section).
+
+#### Notes
+1. The zonegroups section of the object-store-realm CR is only additive. If the playbook is reapplied and zonegroup already exist the zonegroup won't be recreated. If a zonegroup previously created is removed from the list of zonegroups, it will not be deleted. If a new zonegroup is added to the list it will be created. 
```

Comment:
Instead of "playbook is reapplied", how about "operator is restarted"

Reply:
what about "rook operator does not update or delete zone groups"? 
I think that is needed to add a big WARNING in this point, because once we have defined a realm name and zonegroups, they only can be deleted using the toolbox. i think that include in this document a written procedure to do this task would be nice.